SimCalorimetry/EcalTrigPrimProducers/plugins/EcalTrigPrimAnalyzer.cc

```diff
@@ -59,48 +54,32 @@ EcalTrigPrimAnalyzer::EcalTrigPrimAnalyzer(const edm::ParameterSet &iConfig)
   tree_->Branch("ttf", &ttf_, "ttf/I");
   tree_->Branch("fg", &fg_, "fg/I");
   for (unsigned int i = 0; i < 2; ++i) {
-    ecal_et_[i] = new TH1I(ecal_parts_[i].c_str(), "Et", 255, 0, 255);
+    ecal_et_[i] = fs->make<TH1I>(ecal_parts_[i].c_str(), "Et", 255, 0, 255);
     char title[30];
     sprintf(title, "%s_ttf", ecal_parts_[i].c_str());
-    ecal_tt_[i] = new TH1I(title, "TTF", 10, 0, 10);
+    ecal_tt_[i] = fs->make<TH1I>(title, "TTF", 10, 0, 10);
     sprintf(title, "%s_fgvb", ecal_parts_[i].c_str());
-    ecal_fgvb_[i] = new TH1I(title, "FGVB", 10, 0, 10);
+    ecal_fgvb_[i] = fs->make<TH1I>(title, "FGVB", 10, 0, 10);
   }
 
-  recHits_ = iConfig.getParameter<bool>("AnalyzeRecHits");
-  label_ = iConfig.getParameter<edm::InputTag>("inputTP");
   if (recHits_) {
-    hTPvsRechit_ = new TH2F("TP_vs_RecHit", "TP vs  rechit", 256, -1, 255, 255, 0, 255);
-    hTPoverRechit_ = new TH1F("TP_over_RecHit", "TP over rechit", 500, 0, 4);
-    rechits_labelEB_ = iConfig.getParameter<edm::InputTag>("inputRecHitsEB");
-    rechits_labelEE_ = iConfig.getParameter<edm::InputTag>("inputRecHitsEE");
+    hTPvsRechit_ = fs->make<TH2F>("TP_vs_RecHit", "TP vs  rechit", 256, -1, 255, 255, 0, 255);
+    hTPoverRechit_ = fs->make<TH1F>("TP_over_RecHit", "TP over rechit", 500, 0, 4);
     geomToken_ = esConsumes<CaloGeometry, CaloGeometryRecord>();
     endcapGeomToken_ = esConsumes<CaloSubdetectorGeometry, EcalEndcapGeometryRecord>(edm::ESInputTag("", "EcalEndcap"));
     barrelGeomToken_ = esConsumes<CaloSubdetectorGeometry, EcalBarrelGeometryRecord>(edm::ESInputTag("", "EcalBarrel"));
     eTTmapToken_ = esConsumes<EcalTrigTowerConstituentsMap, IdealGeometryRecord>();
   }
 }
 
-EcalTrigPrimAnalyzer::~EcalTrigPrimAnalyzer() {
-  // do anything here that needs to be done at desctruction time
-  // (e.g. close files, deallocate resources etc.)
-
-  histfile_->Write();
-  histfile_->Close();
-}
-
 //
 // member functions
 //
 
 // ------------ method called to analyze the data  ------------
 void EcalTrigPrimAnalyzer::analyze(const edm::Event &iEvent, const edm::EventSetup &iSetup) {
-  using namespace edm;
-  using namespace std;
-
   // Get input
-  edm::Handle<EcalTrigPrimDigiCollection> tp;
-  iEvent.getByLabel(label_, tp);
+  const edm::Handle<EcalTrigPrimDigiCollection> &tp = iEvent.getHandle(tpToken_);
```

Comment:
```suggestion
  const auto &tp = iEvent.get(tpToken_);
```
The Handle is not really needed here as far as I can see. Event::get should work as well and is recommended.

Reply:
done